File: testdata/sample.py
#!/usr/bin/env python3
"""Sample Python module for testing metadata extraction."""

def greet(name):
    """Print a greeting message."""
    print(f"Hello, {name}!")

def add(a, b):
    """Return the sum of two numbers."""
    return a + b

def subtract(a, b):
    """Return the difference of two numbers."""
    return a - b

def multiply(a, b):
    """Return the product of two numbers."""
    return a * b

def divide(a, b):
    """Return the quotient of two numbers."""
    if b == 0:
        raise ValueError("Division by zero")
    return a / b

if __name__ == "__main__":
    name = "World"
    greet(name)
    print(f"2 + 3 = {add(2, 3)}")

File: testdata/sample.go
package main

import (
	"fmt"
	"os"
)

// main is the entry point of the application
func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: program <name>")
		os.Exit(1)
	}

	name := os.Args[1]
	greet(name)
}

// greet prints a greeting message
func greet(name string) {
	fmt.Printf("Hello, %s!\n", name)
}

// add returns the sum of two integers
func add(a, b int) int {
	return a + b
}

// subtract returns the difference of two integers
func subtract(a, b int) int {
	return a - b
}

// multiply returns the product of two integers
func multiply(a, b int) int {
	return a * b
}

// divide returns the quotient of two integers
func divide(a, b int) (int, error) {
	if b == 0 {
		return 0, fmt.Errorf("division by zero")
	}
	return a / b, nil
}
